sherpa/bin/pruned_transducer_statelessX/offline_server.py

```diff
@@ -287,25 +290,20 @@ def __init__(
 
         self.current_active_connections = 0
 
-        assert decoding_method in (
-            "greedy_search",
-            "modified_beam_search",
-        ), decoding_method
         if decoding_method == "greedy_search":
-            nn_and_decoding_func = run_model_and_do_greedy_search
+            self.beam_search = GreedySearchOffline(
+                self.model,
```

Comment:
The CI complains
https://github.com/k2-fsa/sherpa/runs/7479632765?check_suite_focus=true

```
2022-07-23 08:00:57,024 INFO [offline_server.py:258] Using device: cpu
Traceback (most recent call last):
  File "sherpa/bin/pruned_transducer_statelessX/offline_server.py", line 675, in <module>
    main()
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/torch/autograd/grad_mode.py", line [28](https://github.com/k2-fsa/sherpa/runs/7479632765?check_suite_focus=true#step:11:29), in decorate_context
    return func(*args, **kwargs)
  File "sherpa/bin/pruned_transducer_statelessX/offline_server.py", line 643, in main
    num_active_paths=num_active_paths,
  File "sherpa/bin/pruned_transducer_statelessX/offline_server.py", line [29](https://github.com/k2-fsa/sherpa/runs/7479632765?check_suite_focus=true#step:11:30)5, in __init__
    self.model,
AttributeError: 'OfflineServer' object has no attribute 'model'
```

Reply:
I haven't test this part of the code yet. I need to do some extra testing before moving to merging

Reply:
@csukuangfj Seems like everything passed

Reply:
Thanks a lot! 

Merging